fivetran_provider_async/operators.py

```diff
@@ -55,3 +58,60 @@ def execute_complete(self, context: "Context", event: Optional[Dict[Any, Any]] =
                     event["message"],
                 )
                 return event["return_value"]
+
+    def get_openlineage_facets_on_start(self) -> OperatorLineage:
+        """
+        Default extractor method that OpenLineage will call on execute completion.
+        """
+        # Should likely use the sync hook here to ensure that OpenLineage data is
+        # returned before the timeout.
+        hook = self._get_hook()
```

Comment:
This implementation runs even before execute method is run. Does the lineage reflect for you on Marquez or astro-cloud UI? 

There is a known bug on open lineage side with open issue: https://github.com/MarquezProject/marquez/issues/2230

It mightn't be reflecting all values on Marquez UI. WDYT?

Should this not be part of  just `get_openlineage_facets_on_complete()` as its calling `get_openlineage_facets_on_start()` anyways?

Reply:
I haven't verified yet what the graphs look like when _just_ `get_openlineage_facets_on_start()` runs, but it _should_ contain all relevant information, as the input/output datasets being generated are not reliant on the successful completion of a Fivetran run. They are only reliant on the return of the API calls, and all of those values are static after the initial Fivetran sync occurs.

I think it makes more sense to run this with `on_start()`, as it will ensure that the lineage graph is generated and that the lineage graph as a complete artifact with respect to other pipeline elements does not break when a failure occurs. The Alerts panel and the error run facet should handle issues when the Fivetran sync fails. 

I have not tried this with the Marquez UI at all, I've only tested with the Astro UI.